File: backend/validate_env.py
#!/usr/bin/env python3
"""
VEXO Environment Validation Script
Checks if the backend environment is properly configured.
"""

import os
import sys
from pathlib import Path


def check_env_var(
    var_name: str, default_value: str = None, required: bool = False
) -> bool:
    """Check if an environment variable is set."""
    value = os.getenv(var_name)
    status = "✓" if value else ("✗" if required else "⚠")
    color = "\033[92m" if value else ("\033[91m" if required else "\033[93m")
    reset = "\033[0m"

    display_value = (
        value
        if value
        else f"(default: {default_value})" if default_value else "(not set)"
    )
    print(f"  {color}{status}{reset} {var_name}: {display_value}")

    return bool(value) or not required


def check_file_exists(file_path: str, description: str) -> bool:
    """Check if a required file exists."""
    exists = Path(file_path).exists()
    status = "✓" if exists else "✗"
    color = "\033[92m" if exists else "\033[91m"
    reset = "\033[0m"

    print(f"  {color}{status}{reset} {description}: {file_path}")
    return exists


def main():
    """Main validation function."""
    print("🔍 VEXO Backend Environment Validation")
    print("=====================================")

    # Load .env file if it exists
    env_file = Path(".env")
    if env_file.exists():
        print(f"📄 Loading environment from: {env_file.absolute()}")
        # Simple .env file parser
        with open(env_file) as f:
            for line in f:
                line = line.strip()
                if line and not line.startswith("#") and "=" in line:
                    key, value = line.split("=", 1)
                    os.environ[key.strip()] = value.strip()
    else:
        print("⚠  No .env file found, using system environment variables only")

    print()

    # Check API configuration
    print("🌐 API Configuration:")
    all_good = True
    all_good &= check_env_var("API_HOST", "0.0.0.0")
    all_good &= check_env_var("API_PORT", "8000")
    all_good &= check_env_var("API_TITLE", "VEXO Image Validation API")

    print()

    # Check CORS configuration
    print("🔒 CORS Configuration:")
    all_good &= check_env_var("CORS_ORIGINS", "*")
    all_good &= check_env_var("CORS_CREDENTIALS", "true")

    print()

    # Check model configuration
    print("🤖 Model Configuration:")
    all_good &= check_env_var("MODEL_PATH", "vexo_v4_2.keras")
    all_good &= check_env_var("VALIDITY_THRESHOLD", "0.5")

    # Check if model file exists
    model_path = os.getenv("MODEL_PATH", "vexo_v4_2.keras")
    all_good &= check_file_exists(model_path, "Model file")

    print()

    # Check optional configurations
    print("🔧 Optional Configuration:")
    check_env_var("GOOGLE_CREDENTIALS_FILE", "credentials.json")
    check_env_var("LOG_LEVEL", "INFO")
    check_env_var("OCR_CONFIDENCE_THRESHOLD", "0.3")

    print()

    # Check Google Drive credentials if configured
    credentials_file = os.getenv("GOOGLE_CREDENTIALS_FILE", "credentials.json")
    if credentials_file != "credentials.json" or Path(credentials_file).exists():
        print("📁 Google Drive Integration:")
        check_file_exists(credentials_file, "Google credentials")
        print()

    # Summary
    if all_good:
        print("🎉 Environment validation passed! Your backend should be ready to run.")
        print("💡 To start the server: uv run python main.py")
    else:
        print("❌ Environment validation failed. Please check the issues above.")
        print(
            "💡 Make sure you have copied .env.example to .env and configured it properly."
        )
        sys.exit(1)


if __name__ == "__main__":
    main()

File: frontend/validate_env.js
#!/usr/bin/env node
/**
 * VEXO Frontend Environment Validation Script
 * Checks if the frontend environment is properly configured.
 */

import fs from "fs";
import path from "path";

// Colors for console output
const colors = {
  green: "\x1b[32m",
  red: "\x1b[31m",
  yellow: "\x1b[33m",
  blue: "\x1b[34m",
  reset: "\x1b[0m",
};

function checkEnvVar(varName, defaultValue = null, required = false) {
  const value = process.env[varName];
  const status = value ? "✓" : required ? "✗" : "⚠";
  const color = value ? colors.green : required ? colors.red : colors.yellow;

  const displayValue =
    value || (defaultValue ? `(default: ${defaultValue})` : "(not set)");
  console.log(`  ${color}${status}${colors.reset} ${varName}: ${displayValue}`);

  return Boolean(value) || !required;
}

function checkFileExists(filePath, description) {
  const exists = fs.existsSync(filePath);
  const status = exists ? "✓" : "✗";
  const color = exists ? colors.green : colors.red;

  console.log(`  ${color}${status}${colors.reset} ${description}: ${filePath}`);
  return exists;
}

function loadEnvFile(filePath) {
  if (!fs.existsSync(filePath)) {
    return false;
  }

  const content = fs.readFileSync(filePath, "utf8");
  const lines = content.split("\n");

  for (const line of lines) {
    const trimmed = line.trim();
    if (trimmed && !trimmed.startsWith("#") && trimmed.includes("=")) {
      const [key, ...valueParts] = trimmed.split("=");
      const value = valueParts.join("=").trim();
      process.env[key.trim()] = value;
    }
  }

  return true;
}

function main() {
  console.log("🔍 VEXO Frontend Environment Validation");
  console.log("=======================================");

  // Try to load .env.local first, then .env
  const envFiles = [".env.local", ".env"];
  let envLoaded = false;

  for (const envFile of envFiles) {
    if (loadEnvFile(envFile)) {
      console.log(`📄 Loading environment from: ${path.resolve(envFile)}`);
      envLoaded = true;
      break;
    }
  }

  if (!envLoaded) {
    console.log(
      "⚠  No .env.local or .env file found, using system environment variables only"
    );
  }

  console.log();

  // Check API configuration
  console.log("🌐 API Configuration:");
  let allGood = true;
  allGood &= checkEnvVar("NEXT_PUBLIC_API_URL", "http://localhost:8000", true);
  allGood &= checkEnvVar("NEXT_PUBLIC_API_TIMEOUT", "30000");
  allGood &= checkEnvVar("NEXT_PUBLIC_API_RETRY_ATTEMPTS", "3");

  console.log();

  // Check app configuration
  console.log("🎨 App Configuration:");
  checkEnvVar("NEXT_PUBLIC_APP_TITLE", "VEXO Image Validation");
  checkEnvVar(
    "NEXT_PUBLIC_APP_DESCRIPTION",
    "AI-powered image validation and analysis tool"
  );
  checkEnvVar("NEXT_PUBLIC_DEFAULT_THEME", "system");

  console.log();

  // Check feature flags
  console.log("🚀 Feature Flags:");
  checkEnvVar("NEXT_PUBLIC_ENABLE_GOOGLE_DRIVE", "true");
  checkEnvVar("NEXT_PUBLIC_ENABLE_EXCEL_PROCESSING", "true");
  checkEnvVar("NEXT_PUBLIC_ENABLE_BATCH_VALIDATION", "true");
  checkEnvVar("NEXT_PUBLIC_ENABLE_DARK_MODE", "true");

  console.log();

  // Check development settings
  console.log("🛠  Development Settings:");
  checkEnvVar("NEXT_PUBLIC_DEBUG_MODE", "false");
  checkEnvVar("NEXT_PUBLIC_SHOW_DEV_TOOLS", "false");

  console.log();

  // Check essential files
  console.log("📁 Essential Files:");
  allGood &= checkFileExists("package.json", "Package configuration");
  allGood &= checkFileExists("next.config.ts", "Next.js configuration");

  // Check for Tailwind configuration (could be in different places)
  const tailwindConfigs = [
    "tailwind.config.ts",
    "tailwind.config.js",
    "postcss.config.mjs",
  ];
  let tailwindFound = false;
  for (const config of tailwindConfigs) {
    if (fs.existsSync(config)) {
      checkFileExists(config, "Tailwind/PostCSS configuration");
      tailwindFound = true;
      break;
    }
  }
  if (!tailwindFound) {
    console.log(
      `  ${colors.yellow}⚠${colors.reset} Tailwind configuration: Not found (using default config)`
    );
  }

  console.log();

  // Summary
  if (allGood) {
    console.log(
      "🎉 Environment validation passed! Your frontend should be ready to run."
    );
    console.log("💡 To start the development server: bun dev");
  } else {
    console.log(
      "❌ Environment validation failed. Please check the issues above."
    );
    console.log(
      "💡 Make sure you have copied .env.example to .env.local and configured it properly."
    );
    process.exit(1);
  }
}

// Run the main function
main();
